tests/unit_tests/blas/level1/rotmg.cpp

```diff
@@ -137,8 +137,52 @@ int test(device* dev, oneapi::mkl::layout layout) {
     auto x1_accessor = x1_buffer.template get_access<access::mode::read>();
     bool good_x1 = check_equal(x1_accessor[0], x1_ref, error_mag, std::cout);
     auto param_accessor = param_buffer.template get_access<access::mode::read>();
-    bool good_param = check_equal_vector(param_accessor, param_ref, 5, 1, error_mag, std::cout);
-    bool good = good_d1 && good_d2 && good_x1 && good_param;
+
+    constexpr fp unit_matrix = -2;
+    constexpr fp rescaled_matrix = -1;
+    constexpr fp sltc_matrix = 0;
+    constexpr fp clts_matrix = 1;
+
+    fp flag = param_accessor[0];
+    fp h11 = param_accessor[1];
+    fp h12 = param_accessor[3];
+    fp h21 = param_accessor[2];
+    fp h22 = param_accessor[4];
+
+    fp flag_ref = param_ref[0];
+    fp h11_ref = param_ref[1];
+    fp h12_ref = param_ref[3];
+    fp h21_ref = param_ref[2];
+    fp h22_ref = param_ref[4];
+
+    bool flag_good = (flag_ref == flag);
+    bool h11_good = true;
+    bool h12_good = true;
+    bool h21_good = true;
+    bool h22_good = true;
+
+    /* Some values of param have to be ignored depending on the flag value since they are
+     * implementation defined */
```

Comment:
@fabiomestre  Apologies for the delay. Looks like I entered the above comment long time ago, but forgot to finish the review process. Thanks @andrewtbarker for reminding.

Reply:
Thanks for getting back. Rotm and rotmg may look similar but the definition of rotmg happens to be more convoluted and needs to handle many corner cases. I don't remember the details of their definition but one is roughly the inverse of the other.
I've fixed the conflicts now and confirmed that the Rotmg tests are still passing locally.

Reply:
Thanks for feedback. Agreed, after taking a closer look, this case does not apply to `rotm'.

Is this PR needed because SYCL-BLAS backend has different implementation? I guess, I am curious how did you catch this?

Reply:
SYCL-BLAS backend implementation is based on a different reference implementation which can be found in this paper: Hopkins, Tim (1998) Restructuring the BLAS Level 1 Routine for Computing the Modified Givens Transformation. Technical report. , Canterbury, Kent, UK.

The output of the rotmg algorithm is very implementation dependent (i.e. there is multiple correct outputs for the same set of input parameters). So, when porting SYCL-BLAS to oneMKL, we hit cases on this test where there was an expectation that the output of rotmg matches the cblas implementation which is not possible unless the implementation is exactly the same as cblas. So, in the end we decided that it makes more sense to remove the implementation dependent assertions from the test.

Reply:
Thanks for the clarification.